bms-monitor.js

```diff
@@ -1,18 +1,83 @@
-const {expect} = require('chai');
-
-function batteryIsOk(temperature, soc, charge_rate) {
-    if (temperature < 0 || temperature > 45) {
-        console.log('Temperature is out of range!');
-        return false;
-    } else if (soc < 20 || soc > 80) {
-        console.log('State of Charge is out of range!')
-        return false;
-    } else if (charge_rate > 0.8) {
-        console.log('Charge rate is out of range!');
-        return false;
-    }
-    return true;
-}
-
-expect(batteryIsOk(25, 70, 0.7)).to.be.true;
-expect(batteryIsOk(50, 85, 0)).to.be.false;
+// Setting the defautl language English
+let LANGUAGE = 'en';
+
+const PARAMETER_LIMITS = {
+  temperature: {min: 0, max: 45},
+  soc: {min: 20, max: 80},
+  chargeRate: {min: 0, max: 0.8},
+};
+
+const WARNING_TOLERANCE = 0.05;
+
+const WARNING_MESSAGES = {
+  en: {
+    temperature: {
+      low: 'Warning: Approaching lower temperature limit!',
+      high: 'Warning: Approaching upper temperature limit!',
+      error: 'Temperature is out of range!',
+    },
+    soc: {
+      low: 'Warning: Approaching discharge!',
+      high: 'Warning: Approaching charge-peak!',
+      error: 'State of Charge is out of range!',
+    },
+    chargeRate: {
+      low: 'Warning: Approaching minimum charge rate!',
+      high: 'Warning: Approaching maximum charge rate!',
+      error: 'Charge rate is out of range!',
+    },
+  },
+  de: {
+    temperature: {
+      low: 'Warnung: Annäherung an die untere Temperaturgrenze!',
+      high: 'Warnung: Annäherung an die obere Temperaturgrenze!',
+      error: 'Die Temperatur liegt außerhalb des zulässigen Bereichs!',
+    },
+    soc: {
+      low: 'Warnung: Annäherung an Entladung!',
+      high: 'Warnung: Annäherung an Ladungsspitze!',
+      error: 'Ladezustand ist außerhalb des zulässigen Bereichs!',
+    },
+    chargeRate: {
+      low: 'Warnung: Annäherung an minimale Ladegeschwindigkeit!',
+      high: 'Warnung: Annäherung an maximale Ladegeschwindigkeit!',
+      error: 'Laderate liegt außerhalb des zulässigen Bereichs!',
+    },
+  },
+};
+
+const isWithinRange = (value, min, max) => value >= min && value <= max;
+
+function checkWarnings(value, min, max, key) {
+  const tolerance = max * WARNING_TOLERANCE;
+
+  if (value <= min + tolerance) {
+    console.log(WARNING_MESSAGES[LANGUAGE][key].low);
+  } else if (value >= max - tolerance) {
+    console.log(WARNING_MESSAGES[LANGUAGE][key].high);
+  }
+}
+
+function batteryIsOk(temperature, soc, chargeRate) {
+  const parameters = {temperature, soc, chargeRate};
+  let allOk = true;
+
+  for (const [key, value] of Object.entries(parameters)) {
```

Comment:
Nice way to keep the complexity constant even when more parameters are added :+1: